cmd/http/listener.go

```diff
@@ -75,9 +75,10 @@ func isHTTPMethod(s string) bool {
 
 func getResourceHost(bufConn *BufConn, maxHeaderBytes, methodLen int) (resource string, host string, err error) {
 	var data []byte
-	for dataLen := 0; methodLen+dataLen < maxHeaderBytes; dataLen += 8 {
-		if data, err = bufConn.Peek(methodLen + dataLen); err != nil {
-			return "", "", err
+	for dataLen := 0; methodLen+dataLen < maxHeaderBytes; dataLen += 4 {
+		data, _ = bufConn.Peek(methodLen + dataLen)
+		if len(data) == 0 {
+			return resource, host, nil
```

Comment:
if we do this, we shouldn't return error anymore for this function.

Reply:
you are right. updated  PR